src/java/org/apache/cassandra/service/reads/ShortReadPartitionsProtection.java

```diff
@@ -84,6 +84,7 @@ public ShortReadPartitionsProtection(ReadCommand command,
     public UnfilteredRowIterator applyToPartition(UnfilteredRowIterator partition)
     {
         partitionsFetched = true;
+        rangeFetched = true;
```

Comment:
Note: This has been added because it is needed for `MultiRangeReadCommand` and was missing from the original STAR-184 changes.